### ISSUE TITLE
TBA - Episode Title Required - Only for Bulk Season Releases

### ISSUE THREAD

**Issue by thezoggy** [Sonarr/Sonarr#5021]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

tvdb has tba descriptions for bling empire season 2 which came out today

![be1](http://zoggy.net/firefox_2022-05-13_13-40-45.png)

Episode Title Required is set to "Only for Bulk Season Releases", but only episode 2 was being blocked:
![be2](http://zoggy.net/firefox_2022-05-13_13-34-06.png)

![be3](http://zoggy.net/firefox_2022-05-13_13-35-15.png)

### Expected Behavior

I would have expected the whole season to be blocked

### Steps To Reproduce

have series added, wait until air date and releases come out. :P

### Environment

```markdown
- OS: Windows 10
- Sonarr: 3.0.8.1519
- Docker Install: No
- Using Reverse Proxy: No
- Browser: Firefox 100
```

### What branch are you running?

Develop

### Trace Logs?

Let me know what ones you want on discord, 

sonarr.trace.49.txt only goes back to 7:40am today
sonarr.debug.49.txt goes back to yesterday, 12:24pm

### Anything else?

see above

**Comment by markus101** [comment]
Grab the logs for one of the files that imported and logs for one that didn't. Debug logs should be sufficient.

**Comment by thezoggy** [comment]
here you go (debug snippet): https://0bin.net/paste/qlecmz9s#z9tX-K2L8H6F1oZ4onWdkFoc3Rb5cTMDCr1I8yjmW5j

**Comment by markus101** [comment]
I don't see the imports in the linked logs.

**Comment by markus101** [comment]
I've looked into this and this check could fail if the air date for some episodes are missing at the time of import, but I'm not seeing other cases where this might happen. Hopefully the logs show something useful.

**Comment by thezoggy** [comment]
how about this: https://0bin.net/paste/iW6eda+Y#PTxgMXgAl2pHMPsRbSsBHacjIa5y-VW3a1smo67H4kx

**Comment by markus101** [comment]
Yeah, that helps a bit, but unfortunately doesn't 100% explain what happened.

It looks like while you see all the episodes with an airdate in the screenshot, they weren't all present when the file was imported. I've been able to reproduce this with the following example

First episode has air date of 2022-05-13-07:00:00Z
Episodes 2-7 have air date of null

Because they don't have the same air date Sonarr will not see these as airing on the same date as part of a bulk series release. I'm not sure how the episodes were there, but have missing air dates, then later those were filled in, but the titles were still missing, except that maybe the episodes were added recently with a mix of air dates.

The only way to guard against this would be to treat episodes without air dates as airing on the same air date, which will work as long as the first episode has an air date and seems safe enough to block importing if only that episode has an air date.
